addons/web/static/src/utils/dates.js

```diff
@@ -27,100 +87,147 @@ export function formatDateTime(value, options = {}) {
+
 /**
+ * Removes any duplicated alphabetic characters in a given string.
+ * Example: "aa-bb-CCcc-ddD xxxx-Yy-ZZ" -> "a-b-Cc-dD x-Yy-Z"
+ *
  * @param {string} str
  * @returns {string}
  */
 function stripAlphaDupes(str) {
-  // Removes any duplicated alphabetic characters in a given string.
-  // Example: "aa-bb-CCcc-ddD xxxx-Yy-ZZ" -> "a-b-Cc-dD x-Yy-Z"
-  return str.replace(/[a-zA-Z]/g, (letter, index, str) => {
+  return str.replace(alphaRegex, (letter, index, str) => {
     return letter === str[index - 1] ? "" : letter;
   });
 }
 
 /**
- * FIXME: signature looks wrong
+ * Confirms a Luxon DateTime object.
+ * Returns it if within some defined properties.
+ * Returns false otherwise.
  *
  * @param {DateTime | false} date
- * @returns {boolean}
+ * @returns {DateTime | false}
  */
-function check(date) {
-  // FYI, luxon authorizes years until 275760 included...
-  return date.isValid && date.year < 10000 && date;
+function confirm(date) {
+  let valid = date !== false;
+  valid = valid && date.isValid;
+  valid = valid && date.year >= 1000;
+  valid = valid && date.year < 10000;
+  return valid ? date : false;
 }
 
-const nonAlphaRegex = /\W/g;
-const nonDigitsRegex = /\D/g;
-
 /**
- * Utility method to create a Luxon DateTime object.
- * The value can also take the form of a smart date: e.g. "+3w" for three weeks
- * from now. If value can not be parsed with the localized format, the fallback
- * is ISO8601 format.
+ * Parses a string value to an UTC Luxon DateTime object.
  *
- * @param {string} value
- * @param {{format?: string, timezone?: boolean}} options
- * @param {string} [options.format] default value is ISO8601
- * @param {boolean} [options.timezone] parse the date then apply the timezone
- *    offset. Default=false
- * @returns {DateTime|false} Luxon DateTime object
+ * @param {string} value value to parse.
+ *  - Value can take the form of a smart date:
+ *    e.g. "+3w" for three weeks from now.
+ *    (`options.format` and `options.timezone` are ignored in this case)
+ *
+ *  - If value cannot be parsed within the provided format,
+ *    ISO8601 and SQL formats are then tried.
+ *
+ * @param {object} options
+ * @param {boolean} [options.dateOnly]
+ *  The output date will represent the start of day (time is truncated).
+ *  Default=false
+ * @param {string} [options.format]
+ *  Provided format used to parse the input value.
+ *  Default=the session localization format
+ * @param {boolean} [options.timezone]
+ *  - True = input value is considered being in localtime.
+ *  - False = input value is considered being in utc time.
+ *
+ *  NB: ISO strings containing timezone information
+ *      will have priority over this option.
+ *
+ *  Default=false.
+ *
+ * @returns {DateTime|false} Luxon DateTime object (in the UTC timezone)
  */
 export function parseDateTime(value, options = {}) {
   if (!value) {
+    // BOI: is {DateTime|false} really what we want ?
+    // I think we  should return a DateTime.invalid when falsy.
```

Comment:
Good question, that's what we did before. false might be more convenient for the code using it (basically, false is a valid value for a date field, it just means the field is unset). If we always return a date, which might be invalid, the code using it should never forget to check that the validity of the date (but of course, if it can return false, it also has to ensure that the value isn't falsy before manipulating it, so I guess it's no big deal). In any case, when you use a function, you must be aware of its API.

But I agree it would be more elegant.

cc @ged-odoo

Reply:
agreed with @aab-odoo